main.go

```diff
@@ -3,111 +3,86 @@ package main
 import (
 	"flag"
 	"fmt"
-	"io/ioutil"
 	"os"
 
 	"gopkg.in/yaml.v2"
 
 	"github.com/exaring/matroschka-prober/pkg/config"
 	"github.com/exaring/matroschka-prober/pkg/frontend"
-	"github.com/exaring/matroschka-prober/pkg/prober"
-	"github.com/prometheus/client_golang/prometheus"
+	"github.com/exaring/matroschka-prober/pkg/probermanager"
 	log "github.com/sirupsen/logrus"
+	inotify "gopkg.in/fsnotify.v1"
 
 	_ "net/http/pprof"
 )
 
 var (
 	cfgFilepath = flag.String("config.file", "matroschka.yml", "Config file")
-	logLevel    = flag.String("log.level", "debug", "Log Level")
+	logLevel    = flag.String("log.level", "error", "Log Level")
 )
 
 func main() {
 	flag.Parse()
 
 	level, err := log.ParseLevel(*logLevel)
 	if err != nil {
-		log.Errorf("Unable to parse log.level: %v", err)
-		os.Exit(1)
+		log.Fatalf("Unable to parse log.level: %v", err)
 	}
 	log.SetLevel(level)
 
-	cfg, err := loadConfig(*cfgFilepath)
+	w, err := inotify.NewWatcher()
 	if err != nil {
-		log.Errorf("Unable to load config: %v", err)
-		os.Exit(1)
+		log.Fatalf("unable to create inotify watcher: %v", err)
 	}
 
-	confSrc, err := cfg.GetConfiguredSrcAddr()
+	err = w.Add(*cfgFilepath)
 	if err != nil {
-		log.Errorf("Unable to get configured src addr: %v", err)
-		os.Exit(1)
+		log.Fatalf("failed to watch file %q: %v", *cfgFilepath, err)
 	}
 
-	probers := make([]*prober.Prober, 0)
-	for i := range cfg.Paths {
-		for j := range cfg.Classes {
-			log.Infof("Starting prober for path %q class %q", cfg.Paths[i].Name, cfg.Classes[j].Name)
-			p, err := prober.New(prober.Config{
-				BasePort:          *cfg.BasePort,
-				ConfiguredSrcAddr: confSrc,
-				SrcAddrs:          config.GenerateAddrs(*cfg.SrcRange),
-				Hops:              cfg.PathToProberHops(cfg.Paths[i]),
-				StaticLabels:      []prober.Label{},
-				TOS: prober.TOS{
-					Name:  cfg.Classes[j].Name,
-					Value: cfg.Classes[j].TOS,
-				},
-				PPS:                 *cfg.Paths[i].PPS,
-				PayloadSizeBytes:    *cfg.Paths[i].PayloadSizeBytes,
-				MeasurementLengthMS: *cfg.Paths[i].MeasurementLengthMS,
-				TimeoutMS:           *cfg.Paths[i].TimeoutMS,
-			})
-
-			if err != nil {
-				log.Errorf("Unable to get new prober: %v", err)
-				os.Exit(1)
-			}
-
-			err = p.Start()
-			if err != nil {
-				log.Errorf("Unable to start prober: %v", err)
-				os.Exit(1)
-			}
-			probers = append(probers, p)
-		}
+	pm := probermanager.New()
+
+	cfg, err := loadConfig(*cfgFilepath)
+	if err != nil {
+		log.Fatalf("Unable to load config: %v", err)
+	}
+
+	err = pm.Configure(cfg)
+	if err != nil {
+		log.Errorf("reconfiguration failed: %v", err)
```

Comment:
should this also be fatal or is the service able to start up without a configured prober manager?